src/mon/MgrMonitor.cc

```diff
@@ -1303,23 +1314,48 @@ bool MgrMonitor::prepare_command(MonOpRequestRef op)
   } else if (prefix == "mgr module disable") {
     string module;
     cmd_getval(cmdmap, "module", module);
+
+    bool confirmation_flag = false;
+    cmd_getval(cmdmap, "yes_i_really_mean_it", confirmation_flag);
+
     if (module.empty()) {
       r = -EINVAL;
       goto out;
     }
-    if (pending_map.get_always_on_modules().count(module) > 0) {
-      ss << "module '" << module << "' cannot be disabled (always-on)";
-      r = -EINVAL;
-      goto out;
-    }
-    if (!pending_map.module_enabled(module)) {
-      ss << "module '" << module << "' is already disabled";
-      r = 0;
-      goto out;
+    if (module != "volumes") {
```

Comment:
Should this allow disabling _any_ always-on module?

Reply:
It depends on whether or not maintainers of other MGR modules want it.

Reply:
cc @ljflores